core/rootobject/collection.go

```diff
@@ -0,0 +1,294 @@
+// Copyright 2025 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package rootobject
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cockroachdb/errors"
+	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
+	"github.com/dolthub/dolt/go/libraries/doltcore/merge"
+	"github.com/dolthub/dolt/go/store/hash"
+
+	"github.com/dolthub/doltgresql/core/id"
+)
+
+// CollectionFunctions is the collection of functions that facilitate interaction between the root value and a Collection.
+type CollectionFunctions interface {
+	// GetID returns the identifying ID for the associated Collection.
+	GetID() RootObjectID
+	// HandleMerge handles merging of two objects. It is guaranteed that "ours" and "theirs" will not be nil, however
+	// "ancestor" may or may not be nil.
+	HandleMerge(ctx context.Context, mro merge.MergeRootObject) (doltdb.RootObject, *merge.MergeStats, error)
+	// LoadCollection loads the Collection from the given root.
+	LoadCollection(ctx context.Context, root RootValue) (Collection, error)
```

Comment:
This can be removed, I think. What you really need is a substitute for LoadAllCollections, which should be in the new package. That method can simply hard-code all the types of root object collections within itself.

Reply:
After refactoring everything, I think hardcoding makes things a bit more complicated compared to the ID method.